### DISCUSSION TITLE
Upset with 2FA enforcement

### DISCUSSION THREAD

**Discussion by nmontalva** [community/community#78193] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

1. How do we know if this is not a form of collection of phone numbers for marketing purposes?
2. I rather have my github account stolen than be physically assaulted to get my phone. Turning everything to 2FA is a way of promoting crime.
3. 2FA is incredibly annoying while travelling abroad or in places with bad mobile signal.
4. Running out of battery locks you out until you charge your phone.
5. Enforces users to have a smartphone.

**Comment by VIPLightning** [comment]
This is Microsoft's doing and 2FA adds potentially greater security risk. That's pretty much what's happened to 100 million epic games users that were hacked by the Russians. But I digress. I'd switch and migrate all my work to [Codeberg](https://codeberg.org/) or other git websites before December 15 at 0:00 (UTC) arrives and shit hits the fan. Because when it happens, Microsoft pretty much made this website a high valuable target to hit and Microsoft's gonna do what's only in their best interest and not for consumers or developers.

**Reply by telamon** [reply]
![image](https://github.com/community/community/assets/148454/4c9e2d4e-3092-4ebc-8fee-0d64cf31e66a)

Mine says January 19, smells like divide and conquer / stock herding.

Farewell man, and good luck

**Comment by khanson679** [comment]
I also came here to protest mandatory 2FA. It serves no purpose whatsoever for personal projects, does not improve security at all depending on implementation details, and creates many SEVERE problems. The cost-benefit ratio is abysmal.

The fact that mandatory 2FA serves no purpose for small personal projects should be self-evident. But this is actually a major concern, as it increases barriers to entry for beginners. This is terribly ironic in the age of "diversity and inclusion".

The existence of "account recovery methods" undermines the claim that mandatory 2FA increases security. This charade was already unmasked in the context of banking -- once your personal information has been leaked, any criminal can use it to "recover" an account.

Finally, and most importantly, mandatory 2FA contributes to unnecessary reliance on smart phones. If my phone gets broken/lost/stolen or merely runs out of battery, the consequences range from massively inconvenient to completely disastrous, as I will be locked out of everything requiring 2FA. I also resent this personally, as I strive **not** to be glued to my phone for reasons that are well understood by this point, and constantly being forced to get out and unlock my phone for such things makes this nearly impossible. It is also socially harmful, since it means that possession of a working smart phone (specially, Android or iPhone, due to the software duopoly) becomes a prerequisite for participation in society. Once again: "diversity and inclusion".

**Comment by telamon** [comment]
I am also sincerely upset about this.
2FA does not improve security for mundane things. just as studies have shown that passwords degrade in quality on platforms with fervent rotation schemes. 

Dear Github all you will achieve is a mass lockouts, lost authenticators and as OP pointed out, **incentivize** theft, assault and murder.
Basically - general mayhem.

Besides, enforcing additional 2FA on developer platform is quite silly. 
May I remind you that most of us possess an ssh-key? (edit: ssh-keys are VALID 2nd factor auth, why are you demanding a third?)
```bash
$ ssh git@github.com generate-login-url
https://github.com/?nonce=12354
```

But just to clarify, I think Github is in the wrong - you're signaling strong distrust towards your users in an attempt to skirt  responsibilities as a provider.
So the day I can't login, is the day I'll finish [a project](https://github.com/decentlabs-north/git-pit) and say  _"Thank you for all the fish"_.

**Reply by telamon** [reply]
Update: I did some digging and some talking and discovered that besides the central alternatives such as sourcehut, codeberg and gitlab. There's another somewhat biased initiative called [radicle.xyz](https://app.radicle.xyz/nodes/seed.radicle.xyz/rad:z3gqcJUoA1n9HaHKufZs5FCSGazv5) .

I'm still very upset.

**Comment by nmontalva** [comment]
Today was the final day. I had no time to migrate, so I had to enable 2FA.
I have no words to express how annoyed I am to be forced to do something against my will. I'll remember this day whenever I have to work in a basement without mobile signal, whenever I am abroad without roaming, everyday I run out of battery on my phone, I would remember this day dearly if my phone is stolen or lost. Thank you github for ruining my work flow.

**Reply by telamon** [reply]
Today i had a notice 7-days left remaining
Would anybody care to comment why the 2FA lockouts dates differ?

**Comment by DNin01** [comment]
I'm sorry to hear that you're not happy about this, but it is what it is. It is important that contributors use 2FA to protect the software supply chain, [as GitHub says](https://github.blog/2022-05-04-software-security-starts-with-the-developer-securing-developer-accounts-with-2fa/).

If this is bothering you, there are other options. GitHub recommends using an authenticator app on your phone, which does not require you to be connected to cell service or hand over your phone number. As for desktop options, authenticator apps for desktop platforms exist (the only one I know of is Yubico Authenticator which requires a YubiKey, a hardware key brand). There are even authenticator browser extensions out there! There are also desktop password managers that support TOTP (such as KeePassXC), or you can use a passkey if your computer supports the passkeys technology (you can set one up in Password and authentication settings).

**Reply by telamon** [reply]
@DNin01 The page you shared is just marketing a product using "fear" as a propellant..  
2FA is a toy store full of plastic guns in various colours and sizes, it doesn't solve any problems, rather brings hostility to otherwise peaceful environments.

I've spent a couple of months probing the brand you mentioned.. 
Like back at the office we have a box, full of plastic pieces. *sigh* you can have it for free.

**Comment by nmontalva** [comment]
As I knew it was doomed to happen, I needed to use Github on other computer while in a place without mobile or wifi signal... Thank you github for making me waste a day!

**Comment by khanson679** [comment]
I am displeased to report that Github 2FA has failed repeatedly for no discernible reason, using Duo Mobile. Only message is "Two-factor authentication failed. ". SMS 2FA has also failed on at least one occasion. So we can also add to the list of problems: it just doesn't work.
